### ISSUE TITLE
Unable to a11y hyperlink in react-native-web

### ISSUE THREAD

**Issue by kopax-polyconseil** [necolas/react-native-web#2206]
**The problem**

We want to create hyperlink within our desktop header nav menu in a react-native-web application hosted at https://passculture.app

So far, we have tried using the `<Link />` component from react-navigation 6, and using a `<a />` tag without `event.preventDefault()`.

In both case, the application is navigating with page reloading before the event is being called, preventing us to rely on the client routing.

**How to reproduce**

Steps to reproduce:
1. With react-navigation 6, use a `<Link />` and click on it.
2. Otherwise, with a tag `<a />`, prevent the default event to be run and click click on the tag

**Expected behavior**

It should not navigate and reload with the hyperlink

**Environment (include versions). Did this work in previous versions?**

* React Native for Web (version): 0.17.5
* React (version): 17.0.2
* Browser: Version 98.0.4750.0 (Official Build) dev (64-bit)

**Comment by necolas** [comment]
I don't know what this issue is about. There's no test case to try either

**Comment by kopax-polyconseil** [comment]
@necolas `<Link />` component is made to build `<Text />` in react-native with navigation and a `href` generated from the `@react-navigation/native` library.

It work in react-native, but it does not work in `react-native-web` due to react-native-web calling `onClick` after the browser. 

I did a fix here : https://github.com/pass-culture/pass-culture-app-native/pull/2526/files#diff-3a22154ea29db85a13bc4bf9ba3f5bd0cc83403880aa5fc873c3553325f366de

I can make a reproduction for you but you are closing issue so fast that I feel I am losing my time reporting and keep patching library on my own. 

Let me know if you want to dig further and I will add a reproduction.

**Comment by necolas** [comment]
> I can make a reproduction for you but you are closing issue so fast that I feel I am losing my time reporting and keep patching library on my own.

A test case is a requirement for new issues as mentioned in the issue template and contributing guidelines. You didn't respond for 5 days.

If you add a test case that demonstrates the issue I will reopen

**Comment by necolas** [comment]
I don't know why react-navigation is making web so complicated here, so the issue might be in that library. cc @EvanBacon 
 https://reactnavigation.org/docs/use-link-props/

**Comment by kopax-polyconseil** [comment]
Hi @necolas and thanks for your reply, I agree that they did complicate.

I did a reproduction here https://codesandbox.io/s/busy-hill-op9gc?file=/src/App.js 

Unfortunately, the reproduction does not reproduce the error we have in our app, but we didn't change anything related to the navigation binding & events. I assume the size of our app or another library is causing the bug, but we can't remove the code as our app is quite big, and we can't randomly guess. 

We will have to live with the ugly vanillaJS hack on the HTML tag unless someone with the experience of react-navigation or some other library hints us.
